src/foolscap/banana.py

```diff
@@ -346,8 +346,11 @@ def pushSlicer(self, slicer, obj):
         # methods which are *not* generators.
 
         itr = slicer.slice(topSlicer.streamable, self)
-        next = iter(itr).next
-
+        if six.PY2:
+            next = iter(itr).next
+        else:
+            next = iter(itr).__next__
```

Comment:
Oh, I see.. this `__next__` thing wasn't caught everywhere.

A 2/3 compatible idiom is rewriting `x = iter(y).next` as `x = functools.partial(next, iter(y))`.

Reply:
Too much code for the same effect.

Reply:
You think so? I think the if/else is more boilerplate and noisier to read.

Reply:
Are you saying there is a one-liner using `functools.partial` that replaces the entire `if ... else` block ?

Reply:
Ok, I waited for a reply . Closing this as the suggestion is a matter of style of code. Thanks.

Reply:
I wasn't working yesterday.

Yes, the replacement for the whole block is `nextfn = functools.partial(next, iter(itr))` which is 2/3 compatible. Calling it `next` would fail since it would shadow the global.